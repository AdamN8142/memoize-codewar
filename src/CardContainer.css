@import url('https://fonts.googleapis.com/css?family=Indie+Flower');
@import url('https://fonts.googleapis.com/css?family=Molle:400i');
@import url('https://fonts.googleapis.com/css?family=Luckiest+Guy|Molle:400i');


.card-container {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  /* width: 800px; */
  background-image:url("https://wallpapercave.com/wp/gfwPusm.jpg");
  height: 100%;
}

/* .bottom-page {
  /* background-image:url("https://wallpapercave.com/wp/gfwPusm.jpg"); */
/* }  */

