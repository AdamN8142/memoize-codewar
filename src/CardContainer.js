import React, { Component } from 'react';
import Card from './Card'
import './CardContainer.css'


class CardContainer extends Component {
  constructor() {
    super();
  }

  showFilteredCategories = () => {
    let filtered = this.props.filteredCategories.map((codeWar)=> {
      return <Card toggleComplete={this.props.toggleComplete} key={codeWar.id} codeWar={codeWar} completed={this.props.completed}/>
    })
    return filtered
  }


  render() {
    return (
      <div className="bottom-page">
        <div className="card-container">
          {this.showFilteredCategories()}
        </div>
      </div>
    )
  }
}


export default CardContainer;