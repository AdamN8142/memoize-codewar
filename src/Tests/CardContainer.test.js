import React from "react";
import CardContainer from "../CardContainer";
import { shallow } from "enzyme";


describe("CardContainer", () => {
  
  let wrapper;
  const mockProps = {
    toggleComplete: jest.fn(),
    data: [
      {
        "id": 1,
        "name": "Abbreviate a Two Word Name",
        "type": "String",
        "prompt": "Write a function to convert a name into initials. This kata strictly takes two words with one space in between them.",
        "level": "8 kyu",
        "replLink": "https://repl.it/@codesForDays/Abbreviate-a-Two-Word-Name?language=nodejs&folderId=",
        "addedToStorage": false
      },
      {
        "id": 2,
        "name": "Welcome to the City",
        "type": "String",
        "prompt": "Create a method sayHello/say_hello/SayHello that takes as input a name, city, and state to welcome a person. Note that name will be an array consisting of one or more values that should be joined together with one space between each, and the length of the name array in test cases will vary.",
        "level": "8 kyu",
        "replLink": "https://repl.it/@codesForDays/Welcome-to-the-City?language=nodejs&folderId=",
        "addedToStorage": false
      },
        {
        "id": 3,
        "name": "Remove Anchor from URL",
        "type": "String",
        "prompt": "Complete the function/method so that it returns the url with anything after the anchor (#) removed.",
        "level": "7 kyu",
        "replLink": "https://repl.it/@codesForDays/Remove-anchor-from-URL?language=nodejs&folderId=",
        "addedToStorage": false
      }
    ],
    filteredCategories: [
      {
        "id": 1,
        "name": "Abbreviate a Two Word Name",
        "type": "String",
        "prompt": "Write a function to convert a name into initials. This kata strictly takes two words with one space in between them.",
        "level": "8 kyu",
        "replLink": "https://repl.it/@codesForDays/Abbreviate-a-Two-Word-Name?language=nodejs&folderId=",
        "addedToStorage": false
      },
      {
        "id": 2,
        "name": "Welcome to the City",
        "type": "String",
        "prompt": "Create a method sayHello/say_hello/SayHello that takes as input a name, city, and state to welcome a person. Note that name will be an array consisting of one or more values that should be joined together with one space between each, and the length of the name array in test cases will vary.",
        "level": "8 kyu",
        "replLink": "https://repl.it/@codesForDays/Welcome-to-the-City?language=nodejs&folderId=",
        "addedToStorage": false
      },
        {
        "id": 3,
        "name": "Remove Anchor from URL",
        "type": "String",
        "prompt": "Complete the function/method so that it returns the url with anything after the anchor (#) removed.",
        "level": "7 kyu",
        "replLink": "https://repl.it/@codesForDays/Remove-anchor-from-URL?language=nodejs&folderId=",
        "addedToStorage": false
      }
    ],
  filteredCompleted: jest.fn() 
  }
  beforeEach(() => {
    wrapper = shallow(
      <CardContainer
      {...mockProps}
      />
    )
  })

 
  it("should match the snapshot with all data passed in", () => {
      expect(wrapper).toMatchSnapshot()
  })

})