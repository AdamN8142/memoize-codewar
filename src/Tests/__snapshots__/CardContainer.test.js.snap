// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should match the snapshot with all data passed in 1`] = `
<div
  className="bottom-page"
>
  <div
    className="card-container"
  >
    <Card
      codeWar={
        Object {
          "addedToStorage": false,
          "id": 1,
          "level": "8 kyu",
          "name": "Abbreviate a Two Word Name",
          "prompt": "Write a function to convert a name into initials. This kata strictly takes two words with one space in between them.",
          "replLink": "https://repl.it/@codesForDays/Abbreviate-a-Two-Word-Name?language=nodejs&folderId=",
          "type": "String",
        }
      }
      key="1"
      toggleComplete={[MockFunction]}
    />
    <Card
      codeWar={
        Object {
          "addedToStorage": false,
          "id": 2,
          "level": "8 kyu",
          "name": "Welcome to the City",
          "prompt": "Create a method sayHello/say_hello/SayHello that takes as input a name, city, and state to welcome a person. Note that name will be an array consisting of one or more values that should be joined together with one space between each, and the length of the name array in test cases will vary.",
          "replLink": "https://repl.it/@codesForDays/Welcome-to-the-City?language=nodejs&folderId=",
          "type": "String",
        }
      }
      key="2"
      toggleComplete={[MockFunction]}
    />
    <Card
      codeWar={
        Object {
          "addedToStorage": false,
          "id": 3,
          "level": "7 kyu",
          "name": "Remove Anchor from URL",
          "prompt": "Complete the function/method so that it returns the url with anything after the anchor (#) removed.",
          "replLink": "https://repl.it/@codesForDays/Remove-anchor-from-URL?language=nodejs&folderId=",
          "type": "String",
        }
      }
      key="3"
      toggleComplete={[MockFunction]}
    />
  </div>
</div>
`;
